api/app/src/routes/medical/facility.ts

```diff
@@ -0,0 +1,69 @@
+import { Request, Response } from "express";
+import Router from "express-promise-router";
+import { asyncHandler, getCxIdOrFail } from "../util";
+const router = Router();
+import status from "http-status";
+import { Facility } from "../../models/medical/facility";
+import { getOrganizationOrFail } from "../../command/medical/organization/get-organization";
+import { createFacility } from "../../command/medical/facility/create-facility";
+import { updateFacility } from "../../command/medical/facility/update-facility";
+import { getFacilities } from "../../command/medical/facility/get-facility";
+
+/** ---------------------------------------------------------------------------
+ * POST /facility
+ *
+ * Updates or creates the facility if it doesn't exist already.
+ *
+ * @return  {Facility}  The facility.
+ */
+router.post(
+  "/",
+  asyncHandler(async (req: Request, res: Response) => {
+    const cxId = getCxIdOrFail(req);
+
+    // TODO: parse this into model
+    const facilityData = req.body;
+
+    let facility: Facility;
+    if (facilityData.id) {
+      const data = { ...facilityData };
+      delete data.id;
+      facility = await updateFacility({
+        id: facilityData.id,
+        cxId,
+        data,
+      });
+    } else {
+      const org = await getOrganizationOrFail({ cxId });
+      facility = await createFacility({
+        cxId,
+        data: facilityData,
+        organizationNumber: org.organizationNumber,
+      });
+    }
+
+    return res.status(status.OK).json({ id: facility.id, ...facility.data });
```

Comment:
Can we add a TODO to build a DTO here? It would be easy for someone to make a slight change on the return here and not change the `GET` below, for example, while both should be the same for consistency.